RELEASE/scripts/autoscend.ash

```diff
@@ -5258,36 +5254,60 @@ boolean LX_loggingHatchet()
 
 boolean LX_getStarKey()
 {
-	if (internalQuestStatus("questL10Garbage") < 9 || internalQuestStatus("questL11Shen") < 7)
+	//if we don't want the star key then skip this function
+	if(!get_property("auto_getStarKey").to_boolean())
 	{
 		return false;
 	}
-	if(!get_property("auto_getStarKey").to_boolean())
+
+	//if already have the star key or already used it, then stop running this function.
+	if(!needStarKey())
+	{
+		set_property("auto_getStarKey", false);
+		return false;
+	}
+	
+	//if we did not progress enough in giant garbage quest to reach hole in the sky then skip this function
+	if (internalQuestStatus("questL10Garbage") < 9 || internalQuestStatus("questL11Shen") < 7)
```

Comment:
An addendum to your comment, the second part of this condition is so we don't waste turns adventuring in the hole in the sky before Shen Copperhead sends us there (where he sends you is based on the day you first see him hence it's more difficult to sequence in a general enough manner than this slightly brute force method).
Otherwise we could have finished the level 10 quest, farm the star key (if we're not level 11 yet) then start the level 11 quest and spend more turns in the hole in the sky.

Reply:
Oh right, thank you for pointing it out.
fixed on commit f56d9b9